### PR TITLE
[auximg 9/13] Parameterizable image generation for sample sets.

### PR DESCRIPTION
**Pull request by ttung**
Introduce the notion of an ImageFetcher, whose job is to return a FetchedImage for a given FOV/HYB/CH/Z combination.  The goal is to allow us to reuse the logic in `build_sample_experiment.py` for `get_iss_data.py`

`write_experiment_json()` now accepts ImageFetchers for producing an image.  In the case of the `build_sample_experiment.py` script, these will just be configured with DumbImageFetchers, which fetch random data.

Depends on #210, #212

### CODE REVIEW

File: examples/build_sample_experiment.py [hunk 1]
```diff
@@ -135,7 +145,15 @@ def build_image(fov_count, hyb_count, ch_count, z_count, default_shape=(1536, 10
     return collection
 
 
-def write_experiment_json(path, fov_count, hyb_dimensions, aux_name_to_dimensions):
+def write_experiment_json(
```
Comment (ambrosejcarr) [inline examples/build_sample_experiment.py]: Ohhhh, I think I finally get where this is all going. If you have a pile of images with correct names, are you working towards something that will build the `experiment.json` for a user?

File: examples/build_sample_experiment.py [hunk 2]
```diff
@@ -149,14 +167,31 @@ def write_experiment_json(path, fov_count, hyb_dimensions, aux_name_to_dimension
         Dictionary mapping dimension name to dimension size for the hybridization image.
     aux_name_to_dimensions : Mapping[str, Mapping[str, int]]
         Dictionary mapping the auxiliary image type to dictionaries, which map from dimension name to dimension size.
+    hyb_image_fetcher : Optional[ImageFetcher]
```
Comment (ambrosejcarr) [inline examples/build_sample_experiment.py]: This and the two following comments could use a bit more verbosity -- What is an ImageFetcher and why would a user pass it? 

Other things I'm wondering here -- it looks like ImageFetcher looks for single images, but I want to build an experiment! so, how does this work?

File: examples/build_sample_experiment.py [hunk 3]
```diff
@@ -88,7 +93,12 @@ def image_data_handle(self) -> IO:
         return output
 
 
-def build_image(fov_count, hyb_count, ch_count, z_count, default_shape=(1536, 1024)):
+class DumbImageFetcher(ImageFetcher):
```
Comment (ambrosejcarr) [inline examples/build_sample_experiment.py]: I think maybe `RandomNoiseImageFetcher` would be more descriptive.

File: examples/build_sample_experiment.py (hunk 2 above)
Comment (ttung) [reply examples/build_sample_experiment.py]: Added more prose.

**Pull request closed without merging**